pkg/nfd-master/nfd-master.go

```diff
@@ -840,13 +869,24 @@ func (m *nfdMaster) refreshNodeFeatures(nodeName string, labels map[string]strin
 	maps.Copy(extendedResources, crExtendedResources)
 	extendedResources = m.filterExtendedResources(features, extendedResources)
 
+	if m.config.Restrictions.MaxExtendedResourcesPerCR > 0 && len(extendedResources) > m.config.Restrictions.MaxExtendedResourcesPerCR {
```

Comment:
Hmm, I started to think that maybe we could use these settings for disabling extended resources, annotations (or even labels). Or is this a bad idea 🧐 The default would then be -1, meaing "no limit", 0 would in practice mean disable. Thoughts?

Reply:
I think that's better option to start with as an experimental feature and then we can gather insights later on. It would as simple as turning on/off labels, annotations, etc.... But it would give less flexibility/granular control on the behavior of the master compared to the current implementation. wdyt?

Reply:
I agree with Markus here

Reply:
Sorry @AhmedThresh I didn't quite understand your comment/suggestion. One consideration is that we cannot easily (or should not) change the defaults after this gets merged. So we should decide now if we want to do 0=disable

Reply:
fair enough, and for values different than 0 and -1 should they be ignored? e.g.: we don't apply any sort of filtering right?

Reply:
@marquiz looking more into this, I think we should also change the name of these restrictions to enable/disable labels or whatever. We can do this using feature gates. wdyt?

Reply:
@AhmedThresh what do you mean. Change the setting from `MaxExtendedResources: int` to `DisableExtendedResources: bool`? Or adding a separate setting for enabling/disabling, in addition to the max limit?

Reply:
Oh, and feature gates are not for this kind of control. They are supposed to be used for maturizing new/experimental features, eventually each feature gate getting locked to true and finally removed.

Anyway, continuing my previous comment, I think you can turn `Max*` settings to`Disable*` if you want. We can add the max settings later if needed/we want

Reply:
Yes, I meant that we should use Disable/Enable instead of Max. Okay then, there's no need for feature gates.

Reply:
OK. Go ahead with disable/enable if/when that seems better. I also think that's a good initial implementation.